### DISCUSSION TITLE
What if I want to block access to anyone except from reverse-proxy Nginx (Docker vaultwarden)

### DISCUSSION THREAD

**Discussion by cybdn** [dani-garcia/vaultwarden#2277] [Q&A] [answered]
Hi!

I have a vaultwarden running in docker on his own IP with Macvlan... Say for example 192.168.4.10.
I have another container, an nginx, that also has his own IP with Macvlan... say for example 192.168.4.100.

I access Vaultwarden using an hostname that resolves to 192.168.4.100, so the Nginx is handling the connection to the Vaultwarden... Good ol reverse proxy with my SSL certificate :)

Now, everythings works fine!

But... I would like to add a restrictions on the Vaultwarden that accept incoming connection ONLY from the source IP of the reverse proxy... What I want is that every connection to Vaultwarden is first passing through the nginx reverse proxy.
Anyone trying to http connect directly the the Vaultwarden IP will be refused or dropped. 

I connected into the Vaultwarden using exec /bin/bash. I tried to explore the container to find a webserver configurations, an nginx or an apache, anything that I could "control" or configure.
But without success.

So... How can I do that ? How can i restrict access to vaultwarden only from the reverse proxy IP ?

Thanks for you help!

**Comment by redpixy** [comment]
maybe you can try blocking it from your firewall?

**Comment by RealOrangeOne** [comment]
A server-based firewall is probably your best bet. Either that or some other networking tech (eg vpn). Vaultwarden doesn't (and probably shouldn't) have a way of restricting much more than this. The best it can do is bind to a specific network interface.

**Comment by oven8Mitts** [comment, marked as answer]
You may want to use IPTables between docker containers to accomplish this. Assuming you are on Linux.

https://docs.docker.com/network/iptables/

Permit traffic to vaultwarden from anywhere that is related/established. This is so Vaultwarden can contact the internet and download icons and run diagnostics(I turned this off on my setup). Next, permit traffic to vaultwarden port 80 from nginx IP. Finally, drop all other traffic to vaultwarden. Check guides for IPtables for more info. You can filter traffic by interface too, I find this does not work well since interface names in docker are not permanent.

If you do not have Nginx in a docker container, then you may filter with IPTables, and permit the Nginx uid to access the IP, and drop all other connections. I have this set up with Userland Proxy enabled and it works quite well. Traffic would be on the output chain instead of Docker-user chain in this case.

**Reply by cybdn** [reply]
I didn’t know docker was giving that possibility.
That looks like the best answer.
